packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -24,6 +24,65 @@ abstract class AppBarBottomWidget extends Widget {
   double get bottomHeight;
 }
 
+enum _ToolBarSlot {
+  leading,
+  title,
+  actions,
+}
+
+class _ToolBarLayout extends MultiChildLayoutDelegate {
+  _ToolBarLayout({ this.centerTitle });
+
+  // If false the title should be left or right justified within the space bewteen
+  // the leading and actions widgets, depending on the locale's writing direction.
+  // If true the title is centered within the toolbar (not within the horizontal
+  // space bewteen the leading and actions widgets).
+  final bool centerTitle;
+
+  static const double kLeadingWidth = 24.0;
+  static const double kTitleLeft = 64.0; // The AppBar pads left and right an additional 8.0.
+
+  @override
+  void performLayout(Size size) {
+    double actionsWidth = 0.0;
+
+    if (hasChild(_ToolBarSlot.leading)) {
+      final BoxConstraints constraints = new BoxConstraints.tight(new Size(kLeadingWidth, size.height));
+      layoutChild(_ToolBarSlot.leading, constraints);
+      positionChild(_ToolBarSlot.leading, Offset.zero);
+    }
+
+    if (hasChild(_ToolBarSlot.actions)) {
+      final BoxConstraints constraints = new BoxConstraints.loose(size);
+      actionsWidth = layoutChild(_ToolBarSlot.actions, constraints).width;
+      positionChild(_ToolBarSlot.actions, new Offset(size.width - actionsWidth, 0.0));
+    }
+
+    if (hasChild(_ToolBarSlot.title)) {
+      final double maxWidth = size.width - kTitleLeft - actionsWidth;
+      final BoxConstraints constraints = new BoxConstraints.loose(size).copyWith(maxWidth: maxWidth);
+      final Size titleSize = layoutChild(_ToolBarSlot.title, constraints);
+      final double titleY = (size.height - titleSize.height) / 2.0;
+      double titleX = kTitleLeft;
+
+      // If the centered title will not fit between the leading and actions
+      // widgets, then justify it with the wider one.
```

Comment:
Ah, I see. "justify" means [something else](https://en.wikipedia.org/wiki/Typographic_alignment#Justified). I would recommend saying something like "align it against" or "place it abutting" instead of "justify it with".